app/routes/sign-in-oidc.js

```diff
@@ -1,20 +1,20 @@
 const Joi = require('joi')
 const { authConfig } = require('../config')
-const { AUTH_COOKIE_NAME, AUTH_REFRESH_COOKIE_NAME } = require('../constants/cookies')
-const { POST } = require('../constants/http-verbs')
-const { getAccessToken, getRedirectPath } = require('../auth')
-const { decodeState } = require('../auth/defra-id/decode-state')
+const { AUTH_COOKIE_NAME } = require('../constants/cookies')
+const { GET } = require('../constants/http-verbs')
+const { validateState, decodeState, validateInitialisationVector, clearCache, getAccessToken, getRedirectPath } = require('../auth')
+const { cacheRefreshToken } = require('../auth/cache-refresh-token')
 
 module.exports = {
-  method: POST,
+  method: GET,
   path: '/sign-in-oidc',
   options: {
     auth: false,
     plugins: {
       crumb: false
     },
     validate: {
-      payload: Joi.object({
+      query: Joi.object({
```

Comment:
Yes we are.

It's a valid approach that Azure B2C offer.

It's not a risk as the access code can only be used once.  So having it appear in a query string doesn't matter as it can't be replayed later.

We're also verifying the initialisation vector which means the full token exchange transaction can only be performed once.

It's not the same as the scenario we had in the original prototype where we included the access token itself in the url which although many services do use that pattern, I don't believe they should as that is riskier.

But would agree that a POST is inherently more secure than a GET.

The other option we do have is to not use yar for the state and initialisation vector and instead assign our own id on the first request and add that to the cache.  We then include our session id in the state.  That allows us to retrieve it when the response comes back.

Would mean that we'd need handle the refresh token differently and it would be more complicated as we'd be writing our own session handler.  

That may be the right answer, but let me sleep on it.  Will do some more digging into Azure B2C's query option and see what Microsoft say.   Don't want to over engineer if not needed.

Reply:
Slept on it and did some more reading.  As `query` is the default option in Azure B2C and the authorisation code is encrypted and not reusable I'd say leave as is and not add more complexity in managing user sessions

Reply:
Thanks for the explanation. I agree with not over complicating and trusting the Azure B2C structure.